docs/design/2022-03-25-non-transactional-statement.md

```diff
@@ -0,0 +1,123 @@
+# Non-transactional statements
+
+- Author(s): [Ziqian Qin](http://github.com/ekexium)
+- Tracking issue: https://github.com/pingcap/tidb/issues/33485
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+A new syntax that transform a given DML into a serial execution of mutually exclusive collectively exhaustive statements. 
+
+For example, `split on a limit 1000 delete from t` can be transformed into `delete from t where a between 1 and 1000` and `delete from t where a between 1001 and 2000`, assuming the min and max of `a` are respectively 1 and 2000.
+
+## Motivation or Background
+
+Users need to do a bulk delete or update using a single statement, while currently TiDB cannot satisfy the requirement because of the performance issue, transaction size limit, and compatibility issues with tools.
+
+A new syntax is proposed to work around the problem. A non-transactional DML contains a DML and information that are used to "split" the statement, thus it is equivalent to a sequence of DMLs which is not transactional since it does not provide atomicity and probably isolation.
+
+Different from the deprecated batch-DML, a non-transactional DML splits SQLs so every split SQL is a normal statement and does not risk data-index consistency.
+
+## Detailed Design
+
+### Syntax
+
+The syntax: `split on <column_name> limit <batch_size> <DML>`
```

Comment:
Is there a constraint that the split column must be indexed? I suggest to add this constraint. If a user specifies a non-indexed column _by mistake_, the split work itself may require huge amount work.

Reply:
+1, but even with the constraint, the index may not be used in `select` and the paging overhead can be high.

Reply:
Makes sense. I've added the check.

Reply:
> +1, but even with the constraint, the index may not be used in `select` and the paging overhead can be high.

It all relies on the planner.
Do you think it helps to also support dry run the `select` so that users can check the query plan of it?